File: C++/1124.cpp
#include <iostream>
using namespace std;


int main()
{
    int x1, y1, x2, y2, dx, dy;

    cin >> x1 >> y1 >> x2 >> y2;

    dx = x2 - x1;
    dy = y2 - y1;

    if (y1 > 1 && dx == 0 && (dy == 1 || (dy == 2 && y1 == 2)))
        cout << "YES" << endl;
    else
        cout << "NO" << endl;

    return 0;
}

File: Python/1124.py
x1, y1 = map(int, input().split())
x2, y2 = map(int, input().split())

dx = x2 - x1
dy = y2 - y1

if y1 > 1 and dx == 0 and (dy == 1 or dy == 2 and y1 == 2):
    print('YES')
else:
    print('NO')
